### DISCUSSION TITLE
Does rehype-raw work with react-markdown in Firefox.

### DISCUSSION THREAD

**Discussion by rachelrj** [rehypejs/rehype#117] [Q&A] [answered]
I am using rehype-raw as a plugin for react-markdown and it seems to be working in Chrome, but not Firefox.
Chrome: 
<img width="301" alt="Screen Shot 2022-10-25 at 2 06 49 PM" src="https://user-images.githubusercontent.com/4721243/197871562-92c4453b-1992-4482-ba93-13e1bea2e283.png">

Firefox:
<img width="298" alt="Screen Shot 2022-10-25 at 2 06 03 PM" src="https://user-images.githubusercontent.com/4721243/197871642-0d7b7284-f163-4c01-934e-a2e383596842.png">

The code is pretty basic usage. I can show more if necessary.

`<ReactMarkdown rehypePlugins={[rehypeRaw]}>{text}</ReactMarkdown>`

I am using Firefox version 106.0.1 (latest I think)

**Comment by ChristianMurphy** [comment, marked as answer]
Welcome @rachelrj!

Sorry you ran into a spot of trouble.
It does support Firefox when built to run in a browser.
I created an example with CRA (`react-scripts`) as the build tool https://stackblitz.com/edit/github-7ztavd?file=src%2Fapp.tsx and opened it in Firefox 106.0.1
Everything rendered as expected.

<details>
<summary>screenshot</summary>

![Screenshot 2022-10-25 at 13-18-03 Remarkjs -  github - StackBlitz](https://user-images.githubusercontent.com/3107513/197873413-dca58602-9bb4-46f1-8080-9138a9d2af67.png)

</details>

For the issue you're seeing locally, I'd check developer tools in the browser and look for any warning or error messages.
I suspect a build tool may not be configured correctly, but browser logs would likely give more specific idea of what's going wrong.

**Reply by rachelrj** [reply]
Thank you for the quick reply. I will look at the browser logs. I am not seeing any console errors. But I will look into build tools too. Do you know of any specific build tools that have caused people issues?
This project is using SSR with Next.js. Has this caused people issues?
And thank you for the example that you've included.

**Reply by ChristianMurphy** [reply]
It should work in Next.js as well.
Here is a sandbox showing `next` used with `react-markdown` and `rehype-raw` https://stackblitz.com/edit/nextjs-qrwmnc?file=pages%2Findex.js

> Do you know of any specific build tools that have caused people issues?

Older versions of Next (version 11 and earlier) had some issues with ESM, and by extension `react-markdown` and `rehype-react`.
I haven't heard of any major issues popping up using it with Next since version 12 enabled ESM support by default.

> This project is using SSR with Next.js. Has this caused people issues?

Maybe?
Is `react-markdown` being run client side or server side?
I've heard of issues running it fully serverside, because `react-markdown` renders async, and Next expects all server side rendered content to be fully sync.
But if that was happening, nothing should be rendering, rather than a single plugin causing an issue. 🤔 

> I am not seeing any console errors

Hmmm, if not the browser console logs. Anything warning in the next.js logs?

**Reply by rachelrj** [reply]
Thank you again, this was so helpful. We are using version 10 of Next. So that might be the issue. I really do appreciate the help.
